Source/WTF/wtf/Deque.h

```diff
@@ -79,12 +77,12 @@ template<typename T, size_t inlineCapacity> class Deque final {
     
     template<typename U> bool contains(const U&) const;
 
-    T& first() { RELEASE_ASSERT(m_start != m_end); return m_buffer.buffer()[m_start]; }
-    const T& first() const { RELEASE_ASSERT(m_start != m_end); return m_buffer.buffer()[m_start]; }
+    T& first() { return m_buffer.capacitySpan()[m_start]; }
+    const T& first() const { return m_buffer.capacitySpan()[m_start]; }
     T takeFirst();
 
-    T& last() { RELEASE_ASSERT(m_start != m_end); return *(--end()); }
-    const T& last() const { RELEASE_ASSERT(m_start != m_end); return *(--end()); }
+    T& last() { return m_end ? m_buffer.capacitySpan()[m_end - 1] : m_buffer.capacitySpan().back(); }
+    const T& last() const { return m_end ? m_buffer.capacitySpan()[m_end - 1] : m_buffer.capacitySpan().back(); }
```

Comment:
I don't understand how the call to `back()` is useful. Seems that if `m_end` is 0 then this needs to crash or assert. Letting `m_end - 1` under flow would already do that and `back()` might assert or might return an incorrect result.